superset-frontend/src/explore/components/controls/FilterControl/AdhocFilterEditPopoverSimpleTabContent/index.tsx

```diff
@@ -359,7 +359,7 @@ const AdhocFilterEditPopoverSimpleTabContent: React.FC<Props> = props => {
       const col = props.adhocFilter.subject;
       const having = props.adhocFilter.clause === CLAUSES.HAVING;
 
-      if (col && datasource && datasource.filter_select && !having) {
+      if (col && datasource?.filter_select !== false && !having) {
```

Comment:
If we are adding the "filter_select": true in sql_lab.py we can remove this change.  If we keep this change we don't need to make the change in sql_lab.py. 

If we do keep this change, I realized we should actually do a check on datasource because datasource. operations are used within the if block.  so recommend changing to this this:

`if (col && datasource && datasource.filter_select !== false && !having) {`